### DISCUSSION TITLE
[Core Protocol] <TPS and Limits>

### DISCUSSION THREAD

**Discussion by ahmadmardeni1** [ethereum-optimism/developers#586] [Core Protocol]
### Did you check the documentation?

- [X] Yes

### Did you read the specs?

- [X] Yes
- [ ] No

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

What is the max theoretical TPS for a superchain? how did Base reach a 4x recorded and 2x theoretical TPS in comparison to Optimisim Mainnet? would really appreciate if there is a doc with all the tech limits you can share with me 

<img width="931" alt="Screenshot 2024-09-06 at 2 16 30 PM" src="https://github.com/user-attachments/assets/d07a4406-4eaa-4d6c-9d99-cdc035ecc2bb">

**Comment by imtipi** [comment]
no as i know
cc @opfocus @joohhnnn do you guys know?
and also i suggest check gas instead of tps

**Reply by joohhnnn** [reply]
CC @sbvegan

**Reply by opfocus** [reply]
I don't know much about it,
`theoretical TPS` I think it depends on the Block gas target

**Comment by hash1go** [comment]
Hello!
While the "[Standard Rollup Charter](https://gov.optimism.io/t/season-6-draft-standard-rollup-charter/8135/1)" is still in draft stage, looking at the [Responsible Gas Limits](https://gov.optimism.io/t/season-6-draft-standard-rollup-charter/8135#responsible-batch-submission-10) chapter of this document, it states:

> the protocol currently does not implement a hardcoded upper bound on the Gas Limit for a chain.

This means that theoretically, there is no upper limit on TPS, and the appropriate selection of Gas Limit is the responsibility of the chain operator.
Chain operators must be aware that increasing the Gas Limit will raise the hardware requirements for nodes, potentially putting a burden on nodes and other infrastructure.

**Comment by Satyam78612** [comment]
I want to know what you mean by being part of the superchain enables seamless transactions, we will have full control over our chain, scalability, TPS, transaction costs?

**Comment by opfocus** [comment]
https://x.com/jessepollak/status/1846922672980349239    
Regarding base’s adjustment of gas target, you can learn more from this person’s twitter
